### PR TITLE
Support mTLS in gRPC exporters

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed>

Support mutual TLS in gRPC exporters.

Other notable changes:
* TLS config moved to a separate struct
* go field `CertPemFile` renamed to `CaCert`, the config property wasn't changed.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -62,6 +57,26 @@ type GRPCSettings struct {
 	WaitForReady bool `mapstructure:"wait_for_ready"`
 }
 
+// TLSConfig exposes client TLS configuration.
+type TLSConfig struct {
+	// Root CA certificate file for TLS credentials of gRPC client. Should
+	// only be used if `secure` is set to true.
+	CaCert string `mapstructure:"cert_pem_file"`
```
Comment (pavolloffay) [inline config/configgrpc/configgrpc.go:64]: The config property name hasn't been changed.

Can we make a breaking change and change it to include `ca` in name? e.g `ca_cert_pem_file`? Or just `ca_cert`?

Comment (bogdandrutu) [reply config/configgrpc/configgrpc.go:64]: We can probably do that in a separate PR.

This is a breaking change when it comes to the code (not configuration) so you will need to followup with a PR in the collector-contrib where you update otel version (using `make`, see the rule there) and update all the code. Also need to make a quick announcement on the gitter, so if there are others who depend on this they know about this change.

File: config/configgrpc/configgrpc.go [hunk 2]
```diff
@@ -110,3 +125,56 @@ func GrpcSettingsToDialOptions(settings GRPCSettings) ([]grpc.DialOption, error)
 
 	return opts, nil
 }
+
+// Config loads TLS certificates and returns a TLS Config.
+func (c TLSConfig) Config() (*tls.Config, error) {
```
Comment (bogdandrutu) [inline config/configgrpc/configgrpc.go:130]: Maybe `LoadConfig`? or `ToCryptoTlsConfig`

Comment (pavolloffay) [reply config/configgrpc/configgrpc.go:130]: I like more load as it really loads it.

**Comment by pavolloffay** [comment]
@bogdandrutu PR updated. Once this is merged I will open a PR to the contrib to update the code there and announce on the gitter channel.

File: config/configgrpc/configgrpc.go [hunk 3]
```diff
@@ -62,6 +57,26 @@ type GRPCSettings struct {
 	WaitForReady bool `mapstructure:"wait_for_ready"`
 }
 
+// TLSConfig exposes client TLS configuration.
+type TLSConfig struct {
```
Comment (bogdandrutu) [inline config/configgrpc/configgrpc.go:61]: QQ: should this be moved in a package configtls in order to be shared with other protocols that support tls?

Comment (ccaraman) [reply config/configgrpc/configgrpc.go:61]: +1 to this request - I'm working on adding mTLS for receivers/exporters that use HTTP.

**Comment by ccaraman** [comment]
One structural question - There are TLS config options under https://github.com/open-telemetry/opentelemetry-collector/blob/master/receiver/securereceiverconfig.go Should all TLS option then fall under config/...

**Comment by bogdandrutu** [comment]
Going to merge this since we have a separate issue to track the merging of configs. https://github.com/open-telemetry/opentelemetry-collector/issues/933

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (pavolloffay) [reply config/configgrpc/configgrpc.go:64]: @dmitryax updated OTEL collector in the contrib in PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/220/files#diff-993db3ecf423549b9fcb5b7e4d195efeR10. 

That corresponds to https://github.com/open-telemetry/opentelemetry-collector/commit/80ee56387e3473bf0cc70a749441948055cf3682 that landed to master after this PR.

Comment (pavolloffay) [reply config/configgrpc/configgrpc.go:64]: @bogdandrutu ping me if there is anything I can do more. The refactoring/unifying of the TLS spec will be done in #933

**Pull request merged**